app/models/solidus_stripe/gateway.rb

```diff
@@ -85,17 +85,15 @@ def capture(amount_in_cents, payment_intent_id, options = {})
     end
 
     # Authorizes and captures a certain amount on the provided payment source.
-    def purchase(amount_in_cents, _source, options = {})
-      currency = options.fetch(:currency)
-
+    # @todo add support for purchasing custom amounts
+    def purchase(_amount_in_cents, source, options = {})
       # Charge the Customer instead of the card:
-      payment_intent = request do
-        Stripe::PaymentIntent.create({
-          amount: to_stripe_amount(amount_in_cents, currency),
-          currency: currency,
-          **options[:payment_intent_options].to_h
-        })
-      end
+      payment_intent =
+        SolidusStripe::PaymentIntent.create_stripe_intent(
+          payment_method: source.payment_method,
+          order: Spree::Order.find(options[:originator].order_id),
```

Comment:
Needing to provide order and payment method to `create_stripe_intent` in the gateway feel a little bit like broken encapsulation, as it sits (at least in theory) at a lower level. Should this method live on the payment method itself and we merge the gateway to it instead? Thoughts?

Reply:
Yeah, I totally agree. Maybe that could be fixed when we move all API calls to the gateway or an adapter class?

Reply:
The gateway or the payment method are fine (solidus calls the methods on the payment method and are forwarded to the gateway via `delegate`). 

I don't think we need anything fancier than that as Stripe is pretty easy to intercept for testing and probably that's the only place in which you would want to switch implementation, I guess 😊.